src/nvim/getchar.c

```diff
@@ -1721,7 +1721,11 @@ static int vgetorpeek(const int advance)
     } else {
       c = read_readbuffers(advance);
     }
-    if (c != NUL && !got_int) {
+
+    if (got_int) {
+      c = handle_int(advance);
+      exiting = true;
```

Comment:
All cases were covered, but I wanted to be explicit about where the function was exiting. I thought this made it much easier when reasoning about the control flow.
Again, this is a temporary change that makes it easier to verify other commits.